stylus/components/modals.styl

```diff
@@ -9,146 +10,205 @@
+
+    @media (max-width: xxlarge-width + large-margin * 2)
+        width auto
+
+$modal-header
+    @extend $elastic-bar
+    margin 0 0 1rem
+    padding medium-padding 3rem 0 medium-padding
+    font-weight bold
+    overflow visible
+    min-height 2.5rem
+
+    +tiny-screen()
+        margin-bottom .5rem
+        padding small-padding 2rem 0 small-padding
+        font-size 1.25rem
+
+$modal-header--compressed
+    padding small-padding 3rem 0 small-padding
+
+    +tiny-screen()
+        padding tiny-padding 2rem 0 tiny-padding
+
+$modal-header--spacious
+    padding large-padding 3rem 0 large-padding
+
+    +small-screen()
+        padding medium-padding 2rem 0 medium-padding
+
+$modal-content
+    @extend $elastic-content
+    padding 0 medium-padding
+
+    &:last-child
+        padding-bottom medium-padding
+        border-bottom-right-radius .5rem
+        border-bottom-left-radius .5rem
+
+    +tiny-screen()
+        padding 0 small-padding
+
+        &:last-child
+            padding-bottom small-padding
+
+$modal-content--compressed
+    padding 0 small-padding
+
+    &:last-child
+        padding-bottom small-padding
+
+    +tiny-screen()
+        padding 0 tiny-padding
+
+        &:last-child
+            padding-bottom tiny-padding
+
+$modal-content--spacious
+    padding 0 large-padding
+
+    &:last-child
+        padding-bottom large-padding
+
+    +small-screen()
+        padding 0 medium-padding
+
+        &:last-child
+            padding-bottom medium-padding
+
+$modal-footer
+    @extend $elastic-bar
+    padding medium-padding
+    text-align right
+
+    button:last-child
+        margin-right 0
+
+    +small-screen()
+        display flex
+        flex-direction column-reverse
 
         button
-            margin-right  0
-            margin-left   .625rem
-            font-size     .9rem
-
-        +small-screen()
-            &
-                display flex
-                flex-direction column-reverse
-                flex-shrink 0
-                margin-top .5rem
-                button
-                    width 100%
-                    margin-left 0
-                // since the order is reverse we target the first-child
-                button:not(:first-child)
-                    margin-bottom .25rem
-
-        &.coz-modal--hidden
-            top                         -25%
-            opacity                     0
-            transition-timing-function  ease-in
-
-    .coz-modal--fullscreen
-    .coz-modal--overflowHidden
-        overflow hidden
+            width 100%
+            margin 0
+
+            // since the order is reverse we target the first-child
+            &:not(:first-child)
+                margin-bottom .25rem
+
+    +tiny-screen()
+        padding small-padding
+
+$modal-footer--compressed
+    padding small-padding
+
+    +tiny-screen()
+        padding tiny-padding
+
+$modal-footer--spacious
+    padding large-padding
+
+    +tiny-screen()
+        padding medium-padding
```

Comment:
Why do we have to repeat the padding for header, content, and footer ? Could we not just apply it to the entire modal ? It feels like we have a lot of code here.

Reply:
I kept the existing strategy that would prefer adding padding to elements inside the modal than the modal itself.
So when using complex modals you don't have padding by default when you want to take advantage of all available space and also it's easier when a section with a top border should touch the borders and style have the padding inside.

Reply:
OK I see 👍